src/packages/auth/functions.ts

```diff
@@ -28,6 +29,50 @@ export function authorizationHeaderToToken(authorizationHeader: string | undefin
   }
 }
 
+/**
+ * Returns default role, and allowed roles for a user.
+ * If the user does not exist, this function creates the user and gives them the default 'user' role.
+ */
+export async function getUserRoles(username: string): Promise<{ allowed_roles: string[]; default_role: string }> {
+  const db = DbMerlin.getDb();
+
+  const { rows, rowCount } = await db.query(
+    `
+      select hasura_default_role, hasura_allowed_roles
+      from metadata.users_and_roles
+      where username = $1;
+    `,
+    [username],
+  );
+
+  if (rowCount > 0) {
+    const [row] = rows;
+    const { hasura_allowed_roles: allowed_roles, hasura_default_role: default_role } = row;
+    return { allowed_roles, default_role };
+  } else {
+    const default_role = 'user';
+    const allowed_roles = ['user'];
```

Comment:
Adjacent to/Additionally addressing the comment on L69, [I'd proposed that the default allowed roles for a new user would be `[user, viewer]`](https://github.com/NASA-AMMOS/aerie/discussions/983#discussioncomment-6234553). This was done because:
1) A user should be allowed to perform queries as an exclusively more restrictive role than their default.
1) It would satisfy the request for users to be able to enter a 'read-only' mode

Reply:
Are there actual use cases for making (1) and (2) default behavior? They are both still achievable of course if an admin gives a user more roles, but I'm not convinced we need this behavior out of the box.

Reply:
(1) tends to be the precedent with role systems (ie being able to enter `viewer` mode on a Google Doc you're an `editor` of), and without it there's no indication by default that role-switching is a thing and (2) [has been a request for a while](https://github.com/orgs/NASA-AMMOS/projects/2/views/26?filterQuery=is%3Aissue+no%3Asprint+-status%3Adone+read&pane=issue&itemId=16019473).

Reply:
So for (2) I think yes you could use the view role to accomplish it, but I'm not sure that's the best way. It would require setting the viewer role as the default role, which is more far reaching than just locking/unlocking a plan per session.

Reply:
Wouldn't it just require role-swapping for as long as you want to be in "viewer" mode?

Reply:
Yeah it would, but again I don't think we should switch the application-wide role to accomplish just locking/unlocking plan editing. For instance a user role should be able to lock plan while still being allowed to do other things in the app.

Reply:
Thinking about this more - maybe we just make this configurable? So users can set the initial default_role and/or allowed_roles the gateway uses.

Reply:
Updated with new env vars to set the default role and allowed roles in the auth and no auth case. Will update the docs in a later PR.